model/task/task.go

```diff
@@ -852,7 +852,7 @@ func (t *Task) FindTaskOnBaseCommit() (*Task, error) {
 }
 
 func (t *Task) FindTaskOnPreviousCommit() (*Task, error) {
-	return FindOne(db.Query(ByPreviousCommit(t.BuildVariant, t.DisplayName, t.Project, evergreen.RepotrackerVersionRequester, t.RevisionOrderNumber)))
+	return FindOne(db.Query(ByPreviousCommit(t.BuildVariant, t.DisplayName, t.Project, evergreen.RepotrackerVersionRequester, t.RevisionOrderNumber)).Sort([]string{"-" + RevisionOrderNumberKey}))
```

Comment:
Does this sort utilize an index, or is the filtered set small enough that it doesn't need an index? I glanced at the existing indexes in the DB to see which one might fit, but none of them jumped out to me immediately.

Reply:
When I tried performing this build_variant_1_display_name_1_order_1 was the index the query planner used and it didn't have to do any in memory sorts or scan more documents then necessary. 
<img width="1063" alt="image" src="https://github.com/evergreen-ci/evergreen/assets/4605522/9dad0b22-a574-4b21-be64-d08ad7a49591">